andela/travel_tool_back
# #160715637 Checkout of Guest Houses

#### What does this PR do?
This PR contains the implementation for checking out users from a guest house

#### Description of Task to be completed?
- create an endpoint that allows a user update to a checkout status
- write test for the endpoint

#### How should this be manually tested?
- Clone the repo - `git clone https://github.com/andela/travel_tool_back.git`
- Enter into the project directory - `cd travel_tool_back`
- checkout the branch `ft-checkout-guesthouse-160715637`
- Install dependencies - `yarn install`
- Rollback and run database migrations - `yarn db:rollmigrate`
- Start the server - `make start`
- Navigate to [travela](http://travela-local.andela.com:3000/#) in your browser and login
- Login on your browser and grab your token from the application tab in Developers tab
- Open Postman
- Add your token in headers
- Create a guest house using the following [guide](https://github.com/andela/travel_tool_back/pull/65)
- Create a request using the following [guide](https://github.com/andela/travel_tool_back/pull/44) and set yourself as the approver for the request.
- Approve the request as a manager using the following [guide](https://github.com/andela/travel_tool_back/pull/30)
- Make a get request to `localhost:5000/api/v1/trips` to retrieve pending check in records
- Make a put request to `localhost:5000/api/v1/trips/<tripId>` with body object of `{ checkType: "checkIn"}` to check in for a particular trip
- An updated record will be returned with the checkIn time and check status set to 'checked in'
- To checkout a user from the guest house,  Make a put request to `localhost:5000/api/v1/trips/<tripId>` with body object of `{ checkType: "checkOut"}` to check in for a particular trip
- An updated record will be returned with the checkIn time and check status set to 'checked Out'
#### Any background context you want to provide?
Testing this feature requires an existing guesthouse to be present in the database and the request status to be approved.

#### What are the relevant pivotal tracker stories?
[160715637](https://www.pivotaltracker.com/story/show/160715637)

#### Screenshots (if appropriate)
<img width="995" alt="screen shot 2018-10-03 at 11 18 07 am" src="https://user-images.githubusercontent.com/22764463/46404917-7b741c80-c6fe-11e8-8907-3bebb6b03d74.png">

#### Questions:
